services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1000,8 +1002,25 @@ public void addVersionAndStartIngestion(
       int ignoredRmdVersionID,
       boolean versionSwapDeferred,
       int repushSourceVersion) {
-    // Parent controller will always pick the replicationMetadataVersionId from configs.
+    // Parent controller will always pick the replicationMetadataVersionId from
+    // configs.
     final int replicationMetadataVersionId = getRmdVersionID(storeName, clusterName);
+
+    // Ensure valid version
+    if (getStore(clusterName, storeName).isDaVinciPushStatusStoreEnabled()) {
+      Set<Map.Entry<String, Integer>> versionSet = getCurrentVersionsForMultiColos(
+          clusterName,
+          SystemStore.SYSTEM_STORE_NAME_PREFIX + VeniceSystemStoreUtils.DAVINCI_PUSH_STATUS_STORE_STR + "_" + storeName)
```

Comment:
nit: This can be replaced by `VeniceSystemStoreType.getSystemStoreName(storeName)`